cpp/game/board.cpp

```diff
@@ -2639,3 +2639,57 @@ bool Board::isAdjacentToPlaHead(Player pla, Loc loc, Loc plaHead) const {
   );
   return false;
 }
+
+vector<bool> Board::markSymmetricDuplicativeLoc() {
+  vector<bool> isLocSymDuplicative(MAX_ARR_SIZE, false);
+  //this function is the same as in the nninputs.cpp getSymLoc function
+  auto getSymLoc = [this](int x, int y, int symmetry)->Loc{
+    bool transpose = (symmetry & 0x4) != 0;
+    bool flipX = (symmetry & 0x2) != 0;
+    bool flipY = (symmetry & 0x1) != 0;
+    if(flipX) { x = x_size - x - 1; }
+    if(flipY) { y = y_size - y - 1; }
+
+    if(transpose)
+      std::swap(x,y);
+    return Location::getLoc(x,y,transpose ? y_size : x_size);
+  };
+
+  vector<int> symTypes;
+  symTypes.reserve(7);
```

Comment:
Here and elsewhere: do not hardcode 7.
Use the existing NUM_SYMMETRY_COMBINATIONS constant in nninputs.h

Reply:
For the above two comments, originally, I was thinking to use nninputs.h/cpp function. If I do that, it seems that I will break another rule. In your cpp [code review](https://github.com/lightvector/KataGo/tree/master/cpp), you design the codes in a very nice order based on the dependence, core -> game -> neuralnet -> search. That is the reason I didn't use the codes in the neuralnet. I was thinking  that it might be nice to move the getSymLoc function into board.h/cpp. But I am not sure if there are other parts of the codes using getSymLoc.

Reply:
See the comment above - ALL of this code should be moved into the SymmetryHelpers namespace. 

The solution isn't to duplicate code. The solution is to move the code to the correct place.

Reply:
So don't write it in board.h and board.cpp in the first place. You're writing symmetry-related code. Put it with the other symmetry-related code! :)

Reply:
Just followed this suggestion and changes are made  with a new commit.